src/Codeception/Lib/Driver/Facebook.php

```diff
@@ -30,83 +42,107 @@ public function __construct($config, $logCallback = null)
     {
+        $appToken = $this->appId . '|' . $this->appSecret;
         $response = $this->executeFacebookRequest(
-            $this->appSession,
             'POST',
-            '/' . FacebookSession::_getTargetAppId() . '/accounts/test-users',
+            $this->appId . '/accounts/test-users',
+            $appToken,
             [
                 'name' => $name,
                 'installed' => true,
                 'permissions' => $permissions
             ]
-        )->getRawResponse();
+        );
 
-        return json_decode($response, true);
+        return $response->getDecodedBody();
     }
 
     public function deleteTestUser($testUserID)
     {
+        $appToken = $this->appId . '|' . $this->appSecret;
         $this->executeFacebookRequest(
-            $this->appSession,
             'DELETE',
-            '/' . $testUserID
+            '/' . $testUserID,
+            $appToken
         );
     }
 
     public function getTestUserInfo($testUserAccessToken)
     {
         $response = $this->executeFacebookRequest(
-            new FacebookSession($testUserAccessToken),
             'GET',
-            '/me'
-        )->getRawResponse();
+            '/me',
+            $testUserAccessToken
+        );
 
-        return json_decode($response, true);
+        return $response->getDecodedBody();
     }
 
     public function getLastPostsForTestUser($testUserAccessToken)
     {
         $response = $this->executeFacebookRequest(
-            new FacebookSession($testUserAccessToken),
             'GET',
-            '/me/feed'
-        )->getRawResponse();
+            '/me/feed',
+            $testUserAccessToken
+        );
 
-        return json_decode($response, true);
+        return $response->getDecodedBody();
+    }
+
+    public function getVisitedPlaceTagForTestUser($placeId, $testUserAccessToken)
+    {
+        $response = $this->executeFacebookRequest(
+            'GET',
+            "/$placeId",
+            $testUserAccessToken
+        );
+
+        return $response->getDecodedBody();
     }
 
     public function sendPostToFacebook($testUserAccessToken, array $parameters)
     {
         $response = $this->executeFacebookRequest(
-            new FacebookSession($testUserAccessToken),
             'POST',
             '/me/feed',
+            $testUserAccessToken,
             $parameters
-        )->getRawResponse();
-        return json_decode($response, true);
+        );
+        return $response->getDecodedBody();
     }
 
+
     /**
-     * @param FacebookSession $session
      * @param string $method
      * @param string $endpoint
      * @param array $parameters
+     * @param string $token
      * @return \Facebook\FacebookResponse
      */
-    private function executeFacebookRequest(FacebookSession $session, $method, $endpoint, array $parameters = [])
+    private function executeFacebookRequest($method, $endpoint, $token = null, array $parameters = [])
```

Comment:
This is good, actually I though about it earlier but did not do it for some reason, I can do the changes. 

There is something I am not sure about. What do you think about removing haveTestUserLoggedInOnFacebook()? As it seems it is not possible to send post anymore to log in the user here but I had to go to the facebook page and log in by finding the elements. It may be better if user writes this. There are two reasons for this: 
1. The test may not pass all the time but may fail in every 10 runs, I am not sure why this is. 
2. If one is doing acceptance test with WebDriver he will have to add PhpBrowser as dependency even though he will not need PhpBrowser and also haveTestUserLoggedInOnFacebook() will not work by default for the user if the user is using WebDriver.